### PR TITLE
remove testify asserts from pkg/discovery

### PR DESCRIPTION
**Pull request by MHBauer**
for  #16633
Opening PR to get clarity on the mocking situation.

What's left:

```
λ ag testify pkg/
pkg/discovery/kv/kv_test.go
12: "github.com/stretchr/testify/mock"
```

Test's running:

```
+ go test -test.timeout=60m github.com/docker/docker/pkg/discovery
OK: 12 passed
PASS
coverage: 83.8% of statements

+ go test -test.timeout=60m github.com/docker/docker/pkg/discovery/file
OK: 6 passed
PASS
coverage: 93.0% of statements

+ go test -test.timeout=60m github.com/docker/docker/pkg/discovery/kv
OK: 2 passed
PASS
coverage: 89.1% of statements

+ go test -test.timeout=60m github.com/docker/docker/pkg/discovery/nodes
OK: 4 passed
PASS
coverage: 93.8% of statements
```

### CODE REVIEW

File: pkg/discovery/discovery_test.go [hunk 1]
```diff
@@ -3,92 +3,100 @@ package discovery
 import (
 	"testing"
 
-	"github.com/stretchr/testify/assert"
+	. "github.com/go-check/check"
```
Comment (vdemeester) [inline pkg/discovery/discovery_test.go]: I tend to prefer not using `.` as import, and use `check` as is.

Comment (vdemeester) [inline pkg/discovery/discovery_test.go]: There is also a `pkg/checker` that defines additional checkers (and *link* the `go-check` ones)

Comment (MHBauer) [inline pkg/discovery/discovery_test.go]: I will convert the `.` imports.

I do not follow you on the `pkg/checker` part. What would that import look like? Is it preferred?

File: pkg/discovery/nodes/nodes_test.go [hunk 2]
```diff
@@ -4,29 +4,37 @@ import (
...
-func TestInitialize(t *testing.T) {
+// Hook up gocheck into the "go test" runner.
+func Test(t *testing.T) { TestingT(t) }
+
+type DiscoverySuite struct{}
+
+var _ = Suite(&DiscoverySuite{})
+
+func (s *DiscoverySuite) TestInitialize(c *C) {
 	d := &Discovery{}
```
Comment (vdemeester) [inline pkg/discovery/nodes/nodes_test.go]: If you do always the same thing (here `d := &Discovery{}`), you can do something like :

```go
type DiscoverySuite struct{
    d *Discovery
}

func (s *DiscoverySuite) SetUpTest(c *check.C) {
    s.d = &Discovery{}
}
```

File: pkg/discovery/discovery_test.go (hunk 1 above)
Comment (vdemeester) [inline pkg/discovery/discovery_test.go]: e.g.

```go
import (
    "github.com/go-check/check"
    "github.com/docker/docker/pkg/integration/checker"
)

// […]
    c.Assert(err, checker.IsNil)
    c.Assert(myString, checker.Contains, "some substring", check.Commentf("oh.. it failed : %q", myString))
```

**Comment by vdemeester** [comment]
On the mock thingy, there is no rules and no mocking library used in docker.. The less we use mocking library the better actually. So… :sweat_smile: 

I'm still wondering if we should use `go-check` for unit tests too, or only on integration tests. It adds cool stuff but I don't know, need more maintainers opinion on that :smile_cat:.

**Comment by icecrime** [comment]
Thanks so much <3 Only did a superficial review but LGTM assuming green!

**Comment by vieux** [comment]
LGTM

**Comment by vdemeester** [comment]
I guess that answer my questionning :-). Now we neef to figure out about the mock :wink:

**Comment by MHBauer** [comment]
failing test is under investigation in #16598, and besides, unit tests can't make integration tests fail.

**Comment by jfrazelle** [comment]
cant we remove the package

**Comment by jfrazelle** [comment]
this should be minus more :)

**Comment by jfrazelle** [comment]
LGTM

**Comment by jfrazelle** [comment]
ill remove in a pr

**Comment by jfrazelle** [comment]
thanks for this

**Pull request merged**